### PR TITLE
Use ufs.configure, model_configure, MOM_input, data_table, and ice_in templates from UFS-weather-model

### PR DESCRIPTION
**Pull request by aerorahul**
# Description
This PR:
- links `ufs.configure.*.IN`, `model_configure.IN`, `MOM_input_${OCNRES}.in`, `MOM6_data_table.IN`, and `ice_in.IN` from ufs-weather-model into the global-workflow `parm/ufs` directory
- prepares local variables for use in the templates in the respective functions for FV3, MOM6, and CICE6
- uses `atparse.bash` to render these templates into fully formed namelists

- Work was performed by @DeniseWorthen in the ufs-weather-model to templatize several variables that are hard-wired in the ufs-weather-model for CICE. See https://github.com/ufs-community/ufs-weather-model/pull/2010

# Type of change
- New feature - brings one step closer to ufs-weather-model and global-workflow using the same templates
- Maintenance - code refactor

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? YES. The workflow should use the model regression test templates

# How has this been tested?
- Ran C48_S2SW.yaml case on Hera

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: ush/parsing_namelists_CICE.sh
Comment (aerorahul) [inline ush/parsing_namelists_CICE.sh]: @NeilBarton-NOAA 
Can you answer what the appropriate values for these variables are in the workflow?

File: ush/parsing_namelists_CICE.sh
Comment (aerorahul) [inline ush/parsing_namelists_CICE.sh]: @DeniseWorthen 
Can we work to make this (and other `TBT`) variables templated in the ufs-weather-model RTs?

File: ush/parsing_namelists_CICE.sh
Comment (NeilBarton-NOAA) [reply ush/parsing_namelists_CICE.sh]: ### default CICE B-grid, ATM and OCN are provided by cap on A grid
export GRIDATM=A
export GRIDOCN=A
export GRIDICE=B

Comment (aerorahul) [reply ush/parsing_namelists_CICE.sh]: Thanks @NeilBarton-NOAA 
I will set these in the workflow.

Comment (aerorahul) [reply ush/parsing_namelists_CICE.sh]: Included in commit [b7c7ed4](https://github.com/NOAA-EMC/global-workflow/pull/2051/commits/b7c7ed4551febd377b58006856b1d5e2e85c05db)

File: ush/parsing_namelists_CICE.sh
Comment (aerorahul) [reply ush/parsing_namelists_CICE.sh]: This https://github.com/ufs-community/ufs-weather-model/pull/1998 should help.

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul are there run directories available to compare files before and after this change or will I need to make those tests myself?

**Comment by aerorahul** [comment]
> @aerorahul are there run directories available to compare files before and after this change or will I need to make those tests myself?

@JessicaMeixner-NOAA 
Here are the paths for a C48 S2SWA with an `FHMAX=24`:

`develop`: `/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/devtmpl/fcst.65353`
`this PR` : `/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873`

**Comment by JessicaMeixner-NOAA** [comment]
The ice_in had some changes that are potentially of concern: 
* The dumpfreq_n is 1000 in develop and 10000 in this branch.  Hopefully @NeilBarton-NOAA and @DeniseWorthen can clarify if that's actually an issue or not. 
* There are several different output variables that are different between this branch and develop.  I know in the past there was a desire for certain things to be output in regression tests that were not in workflow runs and it appears vice-versa is also true.   I think this needs to be reconciled or made configurable before using the ice_in from ufs-weather-model.

**Comment by DeniseWorthen** [comment]
The variables ``dumpfreq_n`` and ``dumpfreq`` are used in CICE to control when to write restarts. They are not used in the coupled configuration, since we control restart frequency for CMEPS, MOM6 and CICE with ``restart_n`` and ``restart_option``. 

The setttings of ``1000`` and ``d`` from UWM is sets the internal mechanism to write a restart after 1000 days. It should not matter what the setting is, as long as it it >> fhmax.

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh]: Should these be in `config.ice`?

File: ush/parsing_namelists_MOM6.sh
Comment (WalterKolczynski-NOAA) [inline ush/parsing_namelists_MOM6.sh:61]: A lot of commented-out code in this one. Can you clean it up?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Now that we are using templates instead of building a config file, should a lot of these settings be moved out of the parsing scripts into their respective config files?

File: ush/forecast_postdet.sh
Comment (aerorahul) [reply ush/forecast_postdet.sh]: done.

File: ush/parsing_namelists_MOM6.sh
Comment (aerorahul) [reply ush/parsing_namelists_MOM6.sh:61]: done.

**Comment by aerorahul** [comment]
> The ice_in had some changes that are potentially of concern:
> 
> * The dumpfreq_n is 1000 in develop and 10000 in this branch.  Hopefully @NeilBarton-NOAA and @DeniseWorthen can clarify if that's actually an issue or not.
> * There are several different output variables that are different between this branch and develop.  I know in the past there was a desire for certain things to be output in regression tests that were not in workflow runs and it appears vice-versa is also true.   I think this needs to be reconciled or made configurable before using the ice_in from ufs-weather-model.

@JessicaMeixner-NOAA 
Thanks for raising the question.
If there is a desire for the workflow to put out different output, then I would request these variables be templated in the UFSWM and set to their appropriate values here and there.

**Comment by aerorahul** [comment]
@NeilBarton-NOAA @JessicaMeixner-NOAA 
For a quick reference, the differences in `ice_in` are as follows:
```diff
❯ diff -b red_this_pr/ice_in green_develop/ice_in
25c25
<     dumpfreq_n     = 10000
---
>    dumpfreq_n     =  1000
38c38
<     histfreq_n     = 0, 0, 6, 1, 1
---
>    histfreq_n     =  0 , 0 , 6 , 1 , 1
60,62d59
<     grid_atm     = 'A'
<     grid_ocn     = 'A'
<     grid_ice     = 'B'
237,240c234,237
<     f_albice       = 'mdhxx'
<     f_albsno       = 'mdhxx'
<     f_albpnd       = 'mdhxx'
<     f_coszen       = 'mdhxx'
---
>    f_albice        = 'x'
>    f_albsno        = 'x'
>    f_albpnd        = 'x'
>    f_coszen        = 'x'
268c265
<     f_fswthru      = 'mdhxx'
---
>    f_fswthru       = 'x'
286,287c283,284
<     f_divu         = 'x'
<     f_shear        = 'x'
---
>    f_divu          = 'mdhxx'
>    f_shear         = 'mdhxx'
291,294c288,291
<     f_dvidtt       = 'x'
<     f_dvidtd       = 'x'
<     f_daidtt       = 'x'
<     f_daidtd       = 'x'
---
>    f_dvidtt        = 'mdhxx'
>    f_dvidtd        = 'mdhxx'
>    f_daidtt        = 'mdhxx'
>    f_daidtd        = 'mdhxx'
297,298c294,295
<     f_mlt_onset    = 'x'
<     f_frz_onset    = 'x'
---
>    f_mlt_onset     = 'mdhxx'
>    f_frz_onset     = 'mdhxx'
```

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul to my knowledge minus potentially one variable I'm working with Guillaume about, the workflow outputs what we want and not what we don't want and this PR in that view "breaks" that.

**Comment by aerorahul** [comment]
https://cice-consortium-cice.readthedocs.io/en/cice6.0.0.alpha/cice_3_user_guide.html?highlight=icefields_nml#tracers
I am failing to interpret `mdhxx` 
Screenshot 2024-01-02 at 3 20 39 PM
`md` : monthly and daily
`h` : every `histfreq_n` hours
`x` : do not write variable
So, what does `mdhxx` translate to? `monthly`, `daily`, every `histfreq_n` hours, and/or do not write variable.

**Comment by aerorahul** [comment]
@JessicaMeixner-NOAA 
I grepped through the CICE output for some of the variables that are supposedly changed from `x` to `mdhxx` and I don't see them output in `develop` either.
E.g.
```bash
$> ncdump -h /devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc  | grep albice
```
I did a similar search on variables `fswthru`, etc. 
I am surely missing something, but can't put a finger on it.

**Comment by aerorahul** [comment]
Further, I did a `nccmp` compare of the output:
```bash
$> nccmp -d -S -q -f -B --Attribute=checksum --warn=format devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc
```
and I don't see a difference.
It is possible I need to revisit my `nccmp` argument list.

**Comment by DeniseWorthen** [comment]
@aerorahul ``mdhxx``, when combined with ``0,0,6,1,1`` means:

write monthly output every 0 months (ie, don't write)
write daily output every 0 days (ie, don't write)
write hourly output every 6 hours

The last ``xx`` are unused---even though the frequency specifies "each" time interval (ie, every year or every timestep), the "x" turns it off. 

I believe in our case, ice_in could simply specify a freq_n of 6 and then the desired fields would all be just ``h`` instead of ``mhdxx``.

**Comment by DeniseWorthen** [comment]
@aerorahul I think nccmp is giving misleading results. Using (in devtmpl/fcst.65353):

```
cprnc -m CICE_OUTPUT/iceh_06h.2021-03-24-21600.nc /scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-21600.nc
```
 
I see:


```

 Could not find match for file1 variable divu_h in file2
 Could not find match for file1 variable shear_h in file2
 Could not find match for file1 variable dvidtt_h in file2
 Could not find match for file1 variable dvidtd_h in file2
 Could not find match for file1 variable daidtt_h in file2
 Could not find match for file1 variable daidtd_h in file2
 Could not find match for file1 variable mlt_onset_h in file2
 Could not find match for file1 variable frz_onset_h in file2
 Could not find match for file2 variable albice_h in file1
 Could not find match for file2 variable albsno_h in file1
 Could not find match for file2 variable albpnd_h in file1
 Could not find match for file2 variable coszen_h in file1
 Could not find match for file2 variable fswthru_h in file1
```

**Comment by aerorahul** [comment]
> @aerorahul I think nccmp is giving misleading results. Using
> 
> ```
> cprnc -m CICE_OUTPUT/iceh_06h.2021-03-24-21600.nc /scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-21600.nc
> ```
> 
> I see:
> 
> ```
> 
>  Could not find match for file1 variable divu_h in file2
>  Could not find match for file1 variable shear_h in file2
>  Could not find match for file1 variable dvidtt_h in file2
>  Could not find match for file1 variable dvidtd_h in file2
>  Could not find match for file1 variable daidtt_h in file2
>  Could not find match for file1 variable daidtd_h in file2
>  Could not find match for file1 variable mlt_onset_h in file2
>  Could not find match for file1 variable frz_onset_h in file2
>  Could not find match for file2 variable albice_h in file1
>  Could not find match for file2 variable albsno_h in file1
>  Could not find match for file2 variable albpnd_h in file1
>  Could not find match for file2 variable coszen_h in file1
>  Could not find match for file2 variable fswthru_h in file1
> ```

The fault was my use of `nccmp`.  A look at the argument list revealed:
```
❯ nccmp -d -S -f -B  --warn=format /scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc /scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc
DIFFER : Failed to find variable divu_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable shear_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable dvidtt_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable dvidtd_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable daidtt_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable daidtd_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable mlt_onset_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable frz_onset_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/ufstmpl/fcst.224873/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable albice_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable albsno_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable albpnd_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable coszen_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
DIFFER : Failed to find variable fswthru_h in file "/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/devtmpl/fcst.65353/CICE_OUTPUT/iceh_06h.2021-03-24-43200.nc".
```

**Comment by aerorahul** [comment]
A PR in the ufs-weather-model https://github.com/ufs-community/ufs-weather-model/pull/2069 will address the inconsistencies.

**Comment by DeniseWorthen** [comment]
@aerorahul I think I'm confused about what "green_develop" is in your diff. 

Is ``develop`` the GW-develop or UWM-develop? Because it looks like "in this pr" you are adding the field ``f_albice`` for example. That is what the UWM currently uses. But in your UWM PR, you're turning that field off (``f_albice = 'x'``)

**Comment by aerorahul** [comment]
@DeniseWorthen 
red - Values from this PR + defaults/template from UFSWM develop
green - global-workflow develop

Yes, some fields were output in global workflow that were turned off in ufswm and vice-versa.  

The PR in UFSWM makes them consistent.

File: parm/config/gfs/config.ice
Comment (WalterKolczynski-NOAA) [inline parm/config/gfs/config.ice:9]: Since we are changing names anyway, can we make this `CICE_MESH` to be consistent with the others?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: parm/config/gfs/config.ice
Comment (aerorahul) [reply parm/config/gfs/config.ice:9]: `MESH_ICE` is a template variable in `ufs.configure` so we cannot change this variable name.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Comment by aerorahul** [comment]
> ```
> Experiment C96C48_hybatmDA_c22eff4d  *** FAILED *** on Hera
> Experiment C96C48_hybatmDA_c22eff4d  with 1 tasks failed at 01/15/24 04:12:17 AM
> Error logs:
> /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2051/RUNTESTS/COMROOT/C96C48_hybatmDA_c22eff4d/logs/2021122100/gfsfcst.log
> ```

The model is using `gregorian` as the default calendar_type.  Fixed.

**Comment by aerorahul** [comment]
> Experiment C96C48_hybatmDA_30f6095f **SUCCESS** on Hera at 01/16/24 08:04:12 PM

Ok.  This test had previously failed due to the `calender_type` mismatch and was corrected in [30f6095](https://github.com/NOAA-EMC/global-workflow/pull/2051/commits/30f6095fa882d9c019b020d85ddc0fc21658c3d8). It is safe to turn on Orion tests as they are expected to also pass.

**Comment by aerorahul** [comment]
Any final comments on this PR before we merge this today?

**Pull request merged**
